### PR TITLE
passepartui: init at 0.1.3

### LINKED ISSUE #357131: Package request: passepartui
## Project description

A TUI for pass

## Metadata

* homepage URL: https://github.com/kardwen/passepartui
* source URL: https://github.com/kardwen/passepartui
* license: gpl3+ 
* platforms: unix, linux, darwin, ...

image

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by getchoo**
Closes https://github.com/NixOS/nixpkgs/issues/357131

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/passepartui/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/pa/passepartui/package.nix:31]:
```suggestion
    updateScript = nix-update-script {
      extraArgs = [
        "--version-regex"
        "'v(\d+\.\d+\.\d+)'"
      ];
    };
```

It might be worth adding a version regex here? Not sure, i am a a bit inexperienced with update scripts, but might be worth considering.

**Review by LordGrimmauld — COMMENTED** [review]
Tested on x86_64-linux. Builds and runs fine.
HOWEVER: This instantly crashes when started from tty, which i expected to work on a tui app.

```
fetching git input 'git+file:///home/grimmauld/coding/nixpkgs'
[?1049h[?1000h[?1002h[?1003h[?1015h[?1006h[2J[?1049lthread 'main' panicked at src/components/password_store.rs:76:41:
called `Result::unwrap()` on an `Err` value: Unknown { .. } - "Unknown error while interacting with the clipboard: X11 server connection timed out because it was unreachable"
stack backtrace:
   0:     0x55aafecd2dad - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h9754a6437f1de22a
   1:     0x55aafec5460b - core::fmt::write::hb706a393bb60a06f
   2:     0x55aafecdafcf - std::io::Write::write_fmt::h9b447dc5d824d0bd
   3:     0x55aafed077e5 - std::panicking::default_hook::{{closure}}::h227952daede9dd84
   4:     0x55aafed07407 - std::panicking::default_hook::h31626ee1feb8ee2a
   5:     0x55aafed07f38 - std::panicking::rust_panic_with_hook::h76d2aa694a00748e
   6:     0x55aafecd33a7 - std::panicking::begin_panic_handler::{{closure}}::h8855a344ffa1638b
   7:     0x55aafecd2fc9 - std::sys::backtrace::__rust_end_short_backtrace::h3110d0cbfbf26886
   8:     0x55aafed079a4 - rust_begin_unwind
   9:     0x55aafec27f33 - core::panicking::panic_fmt::h3af706d0346c1c60
  10:     0x55aafec27aa6 - core::result::unwrap_failed::h2ba69a02b8a8418b
  11:     0x55aafec7b6a4 - passepartui::components::password_store::PasswordStore::new::h86e3c8e578b0c68d
  12:     0x55aafec97ef0 - passepartui::main::h30da54bf67128974
  13:     0x55aafec92ec3 - std::sys::backtrace::__rust_begin_short_backtrace::h25b594d9507393f9
  14:     0x55aafec8b99c - std::rt::lang_start::{{closure}}::h6d1cb3f06454384e
  15:     0x55aafed0abb3 - std::rt::lang_start_internal::h2084fce485147f46
  16:     0x55aafec9886c - main
  17:     0x7ff6906e727e - __libc_start_call_main
  18:     0x7ff6906e7339 - __libc_start_main_alias_2
  19:     0x55aafec350e5 - _start
  20:                0x0 - <unknown>
```

This might however be an upstream issue expecting some x11 clipboard that just isn't available in tty. I did not dig too deep.

File: pkgs/by-name/pa/passepartui/package.nix
Comment (getchoo) [reply pkgs/by-name/pa/passepartui/package.nix:31]: This basically the default regex for `nix-update`. It should be able to figure out new versions without this

**Review by LordGrimmauld — APPROVED** [review]
Alright, with the tty issue confirmed upstream and the regex apparently not being needed, this looks good to me. Unless you want to wait for the next release upstream to make the cargo patch obsolete too.

**Comment by kardwen** [comment]
I made a new release that should prevent the crash in TTY. Unfortunately pinentry isn't working in TTY, I have to come up with a solution for this.

File: pkgs/by-name/pa/passepartui/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/pa/passepartui/package.nix:28]:
```suggestion
  cargoHash = "sha256-/lgEQ6PmHagt8TlGUV2A95MbV8IQzUwyQ/UkoaGIVHE=";
```

File: pkgs/by-name/pa/passepartui/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/pa/passepartui/package.nix:11]:
```suggestion
  version = "0.1.4";
```

File: pkgs/by-name/pa/passepartui/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/pa/passepartui/package.nix:26]:
```suggestion
```

File: pkgs/by-name/pa/passepartui/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/pa/passepartui/package.nix:17]:
```suggestion
    hash = "sha256-ydX+Rjpfhi0K6f8pzjqWGF0O22gBco6Iot8fXSFNG5c=";
```

**Review by LordGrimmauld — CHANGES_REQUESTED** [review]
Aight, with the update here already, it doesn't make sense to put 0.1.3 into repos, we should go straight for 0.1.4. Especially considering this drops the cargo patch, so the update script doesn't just automagically work.

**Review by LordGrimmauld — APPROVED** [review]
lgtm

File: pkgs/by-name/pa/passepartui/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/pa/passepartui/package.nix:26]:
```suggestion
    description = "TUI for pass, the standard unix password manager";
```

**Review by Aleksanaa — APPROVED** [review]

**Review by matthiasbeyer — APPROVED** [review]
If everyone is fine with this, I'll merge it! Just ping me if you don't have the necessary rights.

**Comment by getchoo** [comment]
Only updated meta.description.

**Review by LordGrimmauld — APPROVED** [review]

**Comment by LordGrimmauld** [comment]
Should the nix package also provide the desktop entry at https://github.com/kardwen/passepartui/blob/main/contrib/passepartui.desktop ?

**Comment by getchoo** [comment]
Personally I don't think it would be worth it as this is a TUI. I would also say that by itself, the desktop file isn't very suitable for Nixpkgs, with it just [executing `foot`](https://github.com/kardwen/passepartui/blob/48f99c22b1f93570cae773540bd1356f52b3d15b/contrib/passepartui.desktop#L8)

**Comment by LordGrimmauld** [comment]
Oh fair point, the foot execution is wild! Good point.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
